src/main/java/io/kabanero/api/restricted/CollectionsEndpoints.java

```diff
@@ -0,0 +1,205 @@
+        UserProfile userProfile = UserProfileManager.getUserProfile();
+        String token = userProfile.getAccessToken();
+
+        String jwt = getJWTFromLogin(user, token, INSTANCE_NAME);
+        if (jwt != null) {
+            NewCookie cookie = new NewCookie(JWT_COOKIE_KEY, jwt);
+            return Response.ok().cookie(cookie).build();
+        }
+        return Response.status(500)
+                .entity(new ResponseMessage("Failed to login to CLI server of instance: " + INSTANCE_NAME)).build();
+    }
+
+    /*
+     * Send a request to the CLI server with the already authenticated GitHub user's
+     * username and OAuth token to get back a JWT we will use to authenticate with
+     * the CLI server
+     * 
+     * @param instanceName the name of the Kabanero instance you wish to login to
+     * 
+     * @return a JWT associated with the logged in GitHub user
+     */
+    private String getJWTFromLogin(String user, String token, String instanceName)
+            throws ClientProtocolException, IOException {
+        CloseableHttpClient client = createTrustAllHttpClientBuilder();
+        // TODO protect against null client
+
+        String cliServerURL = CLI_URL == null ? setCLIURL(INSTANCE_NAME) : CLI_URL;
+        HttpPost httpPost = new HttpPost(cliServerURL + "/login");
+
+        JSONObject gitCreds = new JSONObject();
+        gitCreds.put("gituser", user);
+        gitCreds.put("gitpat", token);
+
+        HttpEntity stringEntity = new StringEntity(gitCreds.toString(), ContentType.APPLICATION_JSON);
+        httpPost.setEntity(stringEntity);
+        httpPost.setHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON);
+        CloseableHttpResponse response2 = client.execute(httpPost);
+
+        try {
+            HttpEntity entity2 = response2.getEntity();
+            String body = EntityUtils.toString(entity2);
+
+            JSONObject jwtJSON = new JSONObject(body);
+
+            String jwt = jwtJSON.getString("jwt");
+            String responseMessage = jwtJSON.getString("message");
+
+            if (jwt == null || responseMessage == null || !"ok".equals(responseMessage)) {
+                LOGGER.log(Level.SEVERE, "Failed to login with the CLI server: " + responseMessage);
+                return null;
+            }
+
+            EntityUtils.consume(entity2);
+            return jwt;
+        } catch (JSONException e) {
+            LOGGER.log(Level.SEVERE, "Failed parsing JSON returned from cli server", e);
+            return null;
+        } finally {
+            response2.close();
+        }
+    }
+
+    private String setCLIURL(String instanceName) {
+        KabaneroInstance wantedInstance = KabaneroManager.getKabaneroManagerInstance()
+                .getKabaneroInstance(instanceName);
+        CLI_URL = wantedInstance.getDetails().getCliURL();
+        return CLI_URL;
+    }
+
+    private static CloseableHttpClient createTrustAllHttpClientBuilder(){
```

Comment:
we will need to update this before the merge into master, this was used for local testing. We should use a secure client for the communication from pod to pod